test/extended/networking/egressip.go

```diff
@@ -0,0 +1,654 @@
+
+			g.By("Determining the target protocol, host and port")
+			targetProtocol, targetHost, targetPort, err = getTargetProtocolHostPort(oc, hasIPv4, hasIPv6, cloudType)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			framework.Logf("Testing against: Protocol %s, TargetHost: %s, TargetPort: %d", targetProtocol, targetHost, targetPort)
+
+			g.By("Creating a namespace for the prober pod")
+			// Create a target namespace and assign source and target namespace
+			// to variables for later use.
+			egressIPNamespace = f.Namespace.Name
+			externalNamespace = oc.SetupNamespace()
+
+			g.By("Getting all worker nodes in alphabetical order")
+			// Get all worker nodes, order them alphabetically with stable
+			// sort order.
+			workerNodesOrdered, err = getWorkerNodesOrdered(clientset)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			for _, s := range workerNodesOrdered {
+				workerNodesOrderedStr = append(workerNodesOrderedStr, s.Name)
+			}
+
+			g.By("Selecting the EgressIP nodes and a non-EgressIP node")
+			// nonEgressIPNode = workerNodesOrdered[0]
+			nonEgressIPNodeStr = workerNodesOrderedStr[0]
+			// egressIPNodesOrdered = workerNodesOrdered[1:]
+			egressIPNodesOrderedStr = workerNodesOrderedStr[1:]
+
+			g.By("Setting the ingressdomain")
+			ingressDomain, err = getIngressDomain(oc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("Setting the EgressIP nodes as EgressIP assignable")
+			for _, node := range egressIPNodesOrderedStr {
+				_, err = oc.AsAdmin().Run("label").Args("node", node, "k8s.ovn.org/egress-assignable=").Output()
+				o.Expect(err).NotTo(o.HaveOccurred())
+			}
+		})
+
+		// Do not check for errors in g.AfterEach as the other cleanup steps will fail, otherwise.
+		g.AfterEach(func() {
+			g.By("Deleting the EgressIP object if it exists")
+			egressIPYamlPath := tmpDirEgressIP + "/" + egressIPYaml
+			if _, err := os.Stat(egressIPYamlPath); err == nil {
+				_, _ = oc.AsAdmin().Run("delete").Args("-f", tmpDirEgressIP+"/"+egressIPYaml).Output()
+			}
+
+			g.By("Removing the EgressIP assignable annotation")
+			for _, node := range egressIPNodesOrderedStr {
+				_, _ = oc.AsAdmin().Run("label").Args("node", node, "k8s.ovn.org/egress-assignable-").Output()
+			}
+
+			g.By("Removing the temp directory")
+			os.RemoveAll(tmpDirEgressIP)
+		})
+
+		g.When("testing to internal targets", func() {
+			g.JustBeforeEach(func() {
+				// Host networked is needed for host networked pods.
+				g.By("Adding SCC hostnetwork to the external namespace")
+				_, err := oc.AsAdmin().Run("adm").Args("policy", "add-scc-to-user", "hostnetwork", fmt.Sprintf("system:serviceaccount:%s:default", externalNamespace)).Output()
+				o.Expect(err).NotTo(o.HaveOccurred())
+			})
+
+			g.It("EgressIP pods should query hostNetwork pods with the local node's SNAT", func() {
+				var targetIP string
+				var targetPort int
+
+				g.By("Selecting a single EgressIP node, and one node per source deployment")
+				// requires a total of 3 worker nodes
```

Comment:
are you only checking > 1, or do you want to explicitly check for 3

Reply:
I just need more than 1 target node (the one hosting the host networked pod) and 2 nodes for the egress ip tests --> a total of 3 worker nodes. I updated the comment in the new version, that'll hopefully be easier to understand.